llvm/lib/Support/YAMLParser.cpp

```diff
@@ -1848,13 +1869,15 @@ bool Scanner::fetchMoreTokens() {
   if (*Current == ',')
     return scanFlowEntry();
 
-  if (*Current == '-' && isBlankOrBreak(Current + 1))
+  if (*Current == '-' && (isBlankOrBreak(Current + 1) || Current + 1 == End))
     return scanBlockEntry();
 
-  if (*Current == '?' && (FlowLevel || isBlankOrBreak(Current + 1)))
+  if (*Current == '?' && (Current + 1 == End || isBlankOrBreak(Current + 1)))
```

Comment:
Ditto here, but also a separate case for the expanded complex-key outside of flow case

Reply:
I agree it is out of scope, at some point you will be fixing the whole world if you continue. Thank you for expanding the test base for the supported cases, it LGTM!